service/history/history_engine.go

```diff
@@ -838,6 +842,21 @@ func (e *historyEngineImpl) NotifyNewHistoryEvent(
 	e.eventNotifier.NotifyNewHistoryEvent(notification)
 }
 
+func (e *historyEngineImpl) GetChasmEngine() chasm.Engine {
+	return e.chasmEngine
+}
+
+func (e *historyEngineImpl) NotifyChasmExecution(executionKey chasm.EntityKey, componentRef []byte) {
+	if e.chasmEngine == nil {
+		return
+	}
+	if chasmEngine, ok := e.chasmEngine.(*ChasmEngine); ok && chasmEngine != nil {
```

Comment:
Why would the engine not be a `ChasmEngine`, either make `NewEngineWithShardContext` accept a `ChasmEngine` or  add this method to the `chasm.Engine` interface.